src/auth.c

```diff
@@ -1716,13 +1716,16 @@ int pr_auth_chroot(const char *path) {
 
   tz = pr_env_get(session.pool, "TZ"); 
   if (tz == NULL) {
+    char tzbuf[32];
+    long off = (timezone / 60 / 60) - daylight;
+    snprintf(tzbuf, sizeof(tzbuf), "%s%ld", daylight ? tzname[1] : tzname[0], off);
```

Comment:
This formatting is making assumptions about the format of the TZ environment variable which are platform-specific.  What is the problematic behavior you're seeing, that makes this patch necessary in your environment?